sklearn/linear_model/_ridge.py

```diff
@@ -1842,13 +1844,42 @@ def fit(self, X, y, sample_weight=None):
         if sample_weight is not None:
             sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
 
-        self.alphas = np.asarray(self.alphas)
-
-        if np.any(self.alphas <= 0):
-            raise ValueError(
-                "alphas must be strictly positive. Got {} containing some "
-                "negative or null value instead.".format(self.alphas)
+        if isinstance(self.alphas, (np.ndarray, list, tuple)):
+            n_alphas = 1 if np.ndim(self.alphas) == 0 else len(self.alphas)
+            if n_alphas != 1:
+                for index, alpha in enumerate(self.alphas):
+                    alpha = check_scalar(
```

Comment:
seeing the call to `check_scalar`, I am think that we could put together the parameters using a `partial`:

```python
check_scalar_alpha = partial(
    scalar, target_type=numbers.Real, min_val=0.0, include_boundaries="neither"
)
# Then each check will be as follows
if n_alphas != 1:
    ...
    check_scalar_alpha(alpha, f"alphas[{index}]")
    ....
    check_scalar_alpha(self.alphas[()], "alpha")
    ...
    # etc.
```

I am under the impression that it will ease readability here because we always check the same parameter indeed.